src/App.tsx

```diff
@@ -1,24 +1,272 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { client } from './utils/fetchClient';
+import { TodoItem } from './components/TodoItem';
+import { Status } from './types/Status';
+import { Filter } from './components/TodoFilter';
 
-const USER_ID = 0;
+const USER_ID = 11498;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [activeFilter, setActiveFilter] = useState(Status.All);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [inputValue, setInputValue] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [wasThereAdding, setWasThereAdding] = useState(false);
+  const [areAllCompletedDeleting, setAreAllCompletedDeleting] = useState(false);
+  const [areAllToggling, setAreAllToggling] = useState(false);
+
+  const completedTodos: Todo[] = [];
+  const activeTodos: Todo[] = [];
+  let visibleTodos: Todo[] = [];
+
+  todos.forEach(todo => {
+    if (todo.completed) {
+      completedTodos.push({ ...todo });
+    } else {
+      activeTodos.push({ ...todo });
+    }
+  });
+
+  switch (activeFilter) {
+    case Status.Active:
+      visibleTodos = activeTodos;
+      break;
+
+    case Status.Completed:
+      visibleTodos = completedTodos;
+      break;
+
+    case Status.All:
+    default:
+      visibleTodos = todos;
+  }
+
+  const areThereCompleted = completedTodos.length !== 0;
+  const areAllCompleted = completedTodos.length === todos.length && todos.length !== 0;
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const errorTimeoutID = useRef(0);
+
+  const handleError = (message: string) => {
+    setHasError(true);
+    setErrorMessage(message);
+    errorTimeoutID.current = window.setTimeout(() => {
+      setHasError(false);
+    }, 3000);
+  };
+
+  useEffect(() => {
+    if (hasError) {
+      setHasError(false);
+    }
+
+    client.get<Todo[]>(`/todos?userId=${USER_ID}`)
+      .then(setTodos)
+      .catch(() => handleError('Unable to load todos'));
+
+    return () => window.clearTimeout(errorTimeoutID.current);
+  }, [activeFilter]);
+
+  useEffect(() => {
+    if (wasThereAdding) {
+      inputRef.current?.focus();
+    }
+  }, [wasThereAdding]);
+
+  const addTodo = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    window.clearTimeout(errorTimeoutID.current);
+
+    if (inputValue.trim() === '') {
```

Comment:
pls see [falsy values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)
```suggestion
    if (!inputValue.trim()) {
```